### ISSUE TITLE
Error in actions.placeBid

### ISSUE THREAD

**Issue by dtome123** [metaplex-foundation/js#155]
There a bug when I call **actions.placeBid** function, it is "Bidder Pot Token Must be a new account". Then I read source code and everything is normal. Why did that happen?
![image](https://user-images.githubusercontent.com/48994663/150625184-8848015b-2d63-43cf-abf6-622803edcd63.png)
                                                                                      _(error)_
![image](https://user-images.githubusercontent.com/48994663/150625307-9584b53f-f024-4432-aaed-f280ddba3f98.png)
                                                                                   _(code)_

**Comment by exromany** [comment]
since this https://github.com/metaplex-foundation/metaplex-program-library/pull/101
```ts
  const bidderPotToken = await AuctionProgram.findProgramAddress([
    Buffer.from(AuctionProgram.PREFIX),
    new PublicKey(bidderPotKey).toBuffer(),
    Buffer.from("bidder_pot_token"),
  ]);
```

**Comment by aheckmann** [comment]
Need to port this fix over https://github.com/metaplex-foundation/metaplex/pull/1557

**Comment by dtome123** [comment]
I try like sugget of exromany then it pass old bug but it have new bug 
![image](https://user-images.githubusercontent.com/48994663/151112281-3d9bf41e-2087-4f4c-a92f-cf235e6ea8ed.png)

**Comment by exromany** [comment]
because simulation fails. you can use `connection.sendRawTransaction` with `skipPreflight` to skip simulation
```ts
connection.sendRawTransaction(tx.serialize(), {
  skipPreflight: true,
});
```

**Comment by dtome123** [comment]
> because simulation fails. you can use `connection.sendRawTransaction` with `skipPreflight` to skip simulation
> 
> ```ts
> connection.sendRawTransaction(tx.serialize(), {
>   skipPreflight: true,
> });
> ```

But after executing, it is still an error transaction so it's not better than. Do you think error from logic code somewhere like smart contract? Because if I use old code in JDK, error is **"Bidder Pot Token must be a new account"**. Then I use your code, it pass but new error  is **"instruction requires an initialized account"**. Is there a conflict here? Thank for your help

**Comment by AmmarKhalid123** [comment]
@basvanberckel  @dtome123  I am facing this same issue, any idea on how to fix this ?

**Comment by basvanberckel** [comment]
I believe this is already enough to fix this error: https://github.com/metaplex-foundation/js/pull/172

**Comment by AmmarKhalid123** [comment]
Yes this seems correct. I hope it gets accepted soon enough

**Comment by AmmarKhalid123** [comment]
@basvanberckel  Is there any way I can use your forked repo to test this solution, and continue on my work until your PR is committed? I have run `npm install git+https://github.com/Beemup/metaplex-js-sdk.git` but unable to import from @metaplex/js, maybe because "lib" folder is missing in the node_modules/@metaplex/js/, which was there when I installed using`npm install @metaplex/js`.

**Comment by basvanberckel** [comment]
What does your import/require statement look like in your app code? You shouldn't need to import from /lib, just from '@metaplex/js'

**Comment by AmmarKhalid123** [comment]
`import { Connection, actions } from '@metaplex/js';`

**Comment by AmmarKhalid123** [comment]
> `import { Connection, actions } from '@metaplex/js';`

On this import, I'm getting error: `Module not found: Can't resolve '@metaplex/js' in 'D:\Office\SolMarketplace\solmarket\src'` . Am I doing something wrong? @basvanberckel

**Comment by basvanberckel** [comment]
Yeah, my bad, you can't directly install from this repo as the artifacts aren't included. 
Clone the repo, build it and install that as a dependency to your own project:

```bash
cd your/project/dir
git clone https://github.com/beemup/metaplex-js-sdk -b placebid-transfer-bug
cd metaplex-js-sdk
npm install
npm run build
cd ..
npm install ./metaplex-js-sdk
```

**Comment by AmmarKhalid123** [comment]
@basvanberckel I'm still getting the same error btw :( Have you tested out this solution on your end?

**Comment by AmmarKhalid123** [comment]
> @basvanberckel I'm still getting the same error btw :( Have you tested out this solution on your end?

Just to clarify, imports are working fine, but I'm getting the initial error of "Bidder pot token must be a new account" on actions.placeBid
@basvanberckel

**Comment by basvanberckel** [comment]
I have my own implementation of placebid that fixes both these errors, the fix linked in this thread above together with my pull request should resolve both. I am not able to write a full test for this right now, there should be a unit test if this needs to be verified in the future.

**Comment by AmmarKhalid123** [comment]
> the fix linked in this thread above together with my pull request should resolve both.

Are you talking about this fix?

> because simulation fails. you can use `connection.sendRawTransaction` with `skipPreflight` to skip simulation
> 
> ```ts
> connection.sendRawTransaction(tx.serialize(), {
>   skipPreflight: true,
> });
> ```

**Comment by basvanberckel** [comment]
The bidder token address should be generated like in https://github.com/metaplex-foundation/metaplex/pull/1557

That will resolve the initial error of this issue.

The transaction order should be changed like in my PR, that will resolve the UninitializedAccount error. It is also an option to remove the revoke instruction entirely, like in https://github.com/metaplex-foundation/metaplex/pull/1557

**Comment by basvanberckel** [comment]
I've updated the pr to fix both errors. @AmmarKhalid123 Would you be able to verify this works in your setup?

**Comment by AmmarKhalid123** [comment]
> I've updated the pr to fix both errors. @AmmarKhalid123 Would you be able to verify this works in your setup?

I will check right away and let you know

**Comment by AmmarKhalid123** [comment]
> I've updated the pr to fix both errors. @AmmarKhalid123 Would you be able to verify this works in your setup?

Works like a charm.

**Comment by AmmarKhalid123** [comment]
@basvanberckel forgot, there was an import of AuctionProgram I remember missing, placeBid.ts file. I got it when tried to build the project, otherwise works great.

**Comment by basvanberckel** [comment]
Ah yes, thanks. Fixed that now
